lib/lockfile.js

```diff
@@ -19,8 +36,21 @@ async function getNewLockfile (packageJson, lock, isNpm) {
         lock
       }
     })
-  } catch (e) {
-    console.log('could not build lockfile', e)
-    throw e
-  }
+      .catch(error => {
+        if (error.statusCode === 404) {
```

Comment:
why retry on `404` specifically? I would think that a `404` is pretty definitive in terms of “this can not be processed” and I would think retrying on network errors like `lib/dbs.js` is what we rather want, but I might be missing something obvious.